File: backend/models/skill.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Skill struct {
	ID          primitive.ObjectID `bson:"_id,omitempty"`
	Name        string
	Category    SkillCategory `bson:"category,omitempty"`
	Description string
}

type SkillCategory string

const (
	CategoryGeneral  SkillCategory = "General"
	CategoryStrength SkillCategory = "Strength"
	CategoryAgility  SkillCategory = "Agility"
	CategoryPassing  SkillCategory = "Passing"
	CategoryMutation SkillCategory = "Mutation"
	CategoryTrait    SkillCategory = "Traits"
)

File: backend/LMWebAPI.Models/Models/Skill.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace LMWebAPI.Models;

public class Skill
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; }
    
    [BsonElement("type")]
    public string Type { get; set; }
    
    [BsonElement("code")]
    public string Code { get; set; }
    
    [BsonElement("name")]
    public string Name { get; set; }
    
    [BsonElement("is_special_trait")]
    public string IsSpecialTrait { get; set; }
    
    [BsonElement("description")]
    public string Description { get; set; }
}